### PR TITLE
Added args and tasks to start and stop the sshx shell

### PR DESCRIPTION
**Pull request by paulotruta**

### CODE REVIEW

File: internal/tasks/tasks.go [hunk 1]
```diff
@@ -240,6 +244,22 @@ func ExecuteTask(task Task) Task {
 			taskResult := taskDisableTunnel()
 			task.Result = sql.NullString{String: taskResult, Valid: true}
 
+		case "start_shell":
+			log.Println("Starting SSHX.io Shell")
+			var args taskStartShellArgs
+			err := json.Unmarshal([]byte(task.Args.String), &args)
+			if err != nil {
+				log.Printf("Error reading arguments or start_shell task: %s", err)
```
Comment (inverse) [inline internal/tasks/tasks.go:252]: Do we need to wrtite back the status of the job?

**Review by inverse — APPROVED** [review]

File: internal/tasks/tasks.go [hunk 2]
```diff
@@ -931,6 +951,78 @@ func taskDisableTunnel() string {
...
 }
 
+func taskStartShell(args taskStartShellArgs) string {
+	fmt.Println("Executing taskStartShell")
+	wsPath := utils.GetPath(utils.WsPath)
+
+	// kill the process if its running
+	utils.Exec(wsPath, "killall", []string{"sshx"})
+
+	cmd := exec.Command("/usr/local/bin/sshx", "--quiet", "--shell", "bash")
```
Comment (inverse) [inline internal/tasks/tasks.go:961]: Where are we installing sshx?

Comment (paulotruta) [reply internal/tasks/tasks.go:961]: We are not doing this in here, as generally the necessary dependency packages are installed using one of the setups (depending on the platform):

- https://github.com/edgebox-iot/multipass-setup (We should update this repo to include the installation of sshx on the setup script)
- https://github.com/edgebox-iot/ua-netinst-config (The same update is needed here)
- https://github.com/edgebox-iot/image-builder (this one is a bit out of date as of now so I guess we can skip adding it here, and we should consider that repo for archival).

I'll be creating PR's to add the sshx installation to the setup scripts.

File: internal/tasks/tasks.go (hunk 1 above)
Comment (paulotruta) [reply internal/tasks/tasks.go:252]: Good catch. Gotta test it going through the error branch, will get back to you on this later ;)

File: internal/tasks/tasks.go [hunk 3]
```diff
@@ -931,6 +951,78 @@ func taskDisableTunnel() string {
...
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		panic(err)
```
Comment (paulotruta) [inline internal/tasks/tasks.go:964]: Just quickly looking at this here... might be worth gracefully handling this 😂

**Pull request merged**